src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -3610,14 +3612,10 @@ private unsafe void ParseRemaining()
                 }
             }
             _info.Offset.End = (ushort)idx;
-        Done:
 
-            cF |= Flags.AllUriInfoSet;
-            lock (_info)
-            {
-                _flags |= cF;
-            }
-            _flags |= Flags.RestUnicodeNormalized;
```

Comment:
Do you know why it was previously written like this?  Seems like the dev went out of their way to set the flags in a non-synchronized manner, when it would have been no more work and no more expensive to just combine this flag into cF above.

Reply:
I tried checking history and it appears this also predates vstfdevdiv. I don't see a good reason as to why it was done like this, but written in this way it's obviously another race condition.

Reply:
> I tried checking history and it appears this also predates vstfdevdiv.

Ok, thanks.

Reply:
@MihaZupan If you want to go further back in history, you can enlist in the Source Depot repos which have the .NET 3.5 and prior sources.